code/controllers/subsystem/tickets/tickets.dm

```diff
@@ -332,18 +336,38 @@ SUBSYSTEM_DEF(tickets)
 	var/ticketCooldown
 	/// Staff member who has assigned themselves to this ticket.
 	var/client/staffAssigned
+	/// The first raw response. Used for ticket conversions
+	var/first_raw_response
+	/// Staff member ckey who took it
+	var/staff_ckey
+	/// The time the staff member took the ticket
+	var/staff_take_time
+	/// List of adminwho data
+	var/list/adminwho_data = list()
+
 
-/datum/ticket/New(tit, raw_tit, cont, num)
+/datum/ticket/New(tit, raw_tit, cont, num, the_ckey)
 	title = tit
 	raw_title = raw_tit
-	content = list()
-	content += cont
-	timeOpened = worldtime2text()
+	client_ckey = the_ckey
+	first_raw_response = cont
+	ticket_responses = list()
+	ticket_responses += new /datum/ticket_response(cont, the_ckey)
+	real_time_opened = SQLtime()
+	ingame_time_opened = (ROUND_TIME ? time2text(ROUND_TIME, "hh:mm:ss") : 0)
 	timeUntilStale = world.time + TICKET_TIMEOUT
 	setCooldownPeriod()
 	ticketNum = num
 	ticketState = TICKET_OPEN
 
+	var/list/this_data = list()
+	for(var/client/C in GLOB.admins)
+		this_data["ckey"] = C.ckey
+		this_data["rank"] = C?.holder.rank
```

Comment:
Probably don't need this since this is a typed loop?
```suggestion
		this_data["rank"] = C.holder.rank
```

Reply:
Nope